frontend/catalyst/programs/verification.py

```diff
@@ -49,6 +63,35 @@ def _verify_nested(
     return state
 
 
+def _verify_observable(obs: Operation, _obs_checker: Callable) -> bool:
+    """Specifies whether or not an observable is accepted by QJITDevice.
+
+    Args: 
+        obs(Operator): the observable to be validated
+        _obs_checker(Callable): a callable that takes an observable
+            and returns True if its supported, and 
+
+    If the observable is built on one or multiple other observables, check
+    both that the overall observable is supported, and that its component
+    parts are supported."""
+
+    if isinstance(obs, Tensor):
+        return all([_verify_observable(o, _obs_checker) for o in obs.obs])
+
+    if isinstance(obs, (Hamiltonian, CompositeOp, SymbolicOp)):
+
+        _obs_checker(obs)
```

Comment:
There seems to be some inconsistencies around what `_obs_checker` does. In the doc it is stated that `_obs_checker` should return `True` if the observable is supported, but in reality it does nothing if the observable is supported and raises an error if it isn't, which makes it so that `_verify_observable` would also raise an error instead of returning a `bool`.

Reply:
Good catch, thanks! I initially had a `stopping_condition` like in PL, but decided to switch to `_obs_checker` to mimic the existing transforms for validation in catalyst instead, and didn't update the documentation to reflect that.

Reply:
all updated now